### ISSUE TITLE
No Rest API found. Caching settings will not be updated. when using shared API GW

### ISSUE THREAD

**Issue by mzmiric5** [DianaIonita/serverless-api-gateway-caching#16]
There seems to be an issue with the plugin when attempting to use it with the shared API Gateway.

It looks like when using a shared API GW, defined through 
```yaml
provider:
  apiGateway:
    restApiId:
      'Fn::ImportValue': CfOutputRefFromAnotherStack
    restApiRootResourceId:
      'Fn::ImportValue': CfOutputRefFromAnotherStack
```
the plugin throws the `No Rest API found. Caching settings will not be updated.` error.

This seems to be caused by the following code:
https://github.com/DianaIonita/serverless-api-gateway-caching/blob/166bd2e0331ccfa1706165ff2a514b773c5f2033/src/apiGatewayCachingPlugin.js#L50

While the plugin seems to be referencing
`get(serverless, 'service.provider.apiGateway.restApiId');`
to get a custom API GW id, it does not seem to pass the validation (as SLS does not add an external API GW to the `service.provider.compiledCloudFormationTemplate.Resources`). Manually overiding the `restApiExists` function to return `true` successfully deploys the functions and correctly enables caching.

In addition (and this is probably out of scope of this issue and the plugin), but it would be nice to be able to apply caching settings to the individual functions without having to define the caching settings for the gateway (for uses cases where you have functions in many different serverless.yml files that you want to add caching to and an API GW defined in another service/stack).

**Comment by DianaIonita** [comment]
Hi @mzmiric5,
I've reproduced the issue you're having, however, I can't get my code to a state where `Manually overiding the restApiExists function to return true successfully deploys the functions and correctly enables caching`. It fails for me with this error message: `The CloudFormation template is invalid: Unresolved resource dependencies [ApiGatewayRestApi] in the Outputs block of the template`.
Do you have a sample `serverless.yml` file (or two, if you're referencing a Rest API from another project) I could take a look at and start debugging?

Thanks

**Comment by mzmiric5** [comment]
Hey @DianaIonita, thanks for looking into this. We were pretty much in a rush to get this working with our project, so we ended up just making a fork with a quick patch branch which worked for us https://github.com/doubly/serverless-api-gateway-caching/tree/skip-restapi-check . This was really just a band-aid solution where we added `skipRestAPICheck` as a custom option that allowed us to ship this to production.

Since our project is not open source, I've created a minimum needed setup repo which is using our fork and branch of the plugin and deploys successfully https://github.com/mzmiric5/sls-api-gw-caching-remoteAPIRef-setup . Let me know if this works for you, and if we can help debug this issue more.

**Comment by DianaIonita** [comment]
Hey @mzmiric5,
Thank you for the sample code you provided, it made it really easy to see where the issue was.
I just released a new version of the package, [1.1.6](https://github.com/DianaIonita/serverless-api-gateway-caching/releases/tag/1.1.6), which first checks the yaml for a pre-defined `restApiId`.
Please let me know if that solves your problem.

**Comment by mzmiric5** [comment]
Hey @DianaIonita that seems to have resolved the issue. Thanks :)
